src/main/java/org/spongepowered/api/data/MemoryDataView.java

```diff
@@ -844,9 +844,39 @@ public boolean equals(@Nullable Object obj) {
             return false;
         }
         final MemoryDataView other = (MemoryDataView) obj;
+        if (!Objects.equal(this.path, other.path) || this.map.size() != other.map.size()) {
+            return false;
+        }
+        for (Map.Entry<String, Object> entry : this.map.entrySet()) {
+            if (!equalsCheckArrays(entry.getValue(), other.map.get(entry.getKey()))) {
```

Comment:
This doesn't account for `other.map.get(entry.getKey())` returning null and is duplicating what's inside `Arrays#deepEquals`.
This check can be reduced to `if (!Objects.deepEquals(entry.getValue(), other.map.get(entry.getKey())))`

Reply:
Agreed, a lot of the deep equals can be replacing some of the equality methods written here. Plus, this class being deprecated, will need an update to the SpongeCommon variation.